SofaKernel/modules/SofaHelper/src/sofa/helper/system/FileRepository.cpp

```diff
@@ -76,37 +76,44 @@ std::string cleanPath( const std::string& path )
 
 // Initialize PluginRepository and DataRepository
 #ifdef WIN32
-FileRepository PluginRepository( "SOFA_PLUGIN_PATH", Utils::getExecutableDirectory().c_str() );
+FileRepository PluginRepository(
+        "SOFA_PLUGIN_PATH", {
+            Utils::getExecutableDirectory(),
+            Utils::getSofaPathTo("plugins")
+        }
+);
 #else
-FileRepository PluginRepository( "SOFA_PLUGIN_PATH", Utils::getSofaPathTo("lib").c_str() );
+FileRepository PluginRepository(
+        "SOFA_PLUGIN_PATH", {
+            Utils::getSofaPathTo("lib"),
+            Utils::getSofaPathTo("plugins"),
+        }
+);
 #endif
-FileRepository DataRepository( "SOFA_DATA_PATH", 0, {
+FileRepository DataRepository( "SOFA_DATA_PATH", nullptr, {
                                    { Utils::getSofaPathTo("etc/sofa.ini"), {"SHARE_DIR", "EXAMPLES_DIR"} }
                                });
 
-
-FileRepository::FileRepository(const char* envVar, const char* relativePath, const fileKeysMap& iniFilesAndKeys)
-{
+FileRepository::FileRepository(const char* envVar, const std::vector<std::string> & paths, const fileKeysMap& iniFilesAndKeys) {
```

Comment:
initializer_list instead of vector ?

Reply:
So, the main difference here (beside the copy overhead, which is equivalent for both of these in c++<20), is that `std::vector` allows to call the constructor at runtime with a dynamic list of paths (a list filled during runtime), while `std::initializer_list` is compile time only.  I'll let it at `std::vector` even if we are only using the compile time initialization, just in case someone wants to dynamically create a FileRepository at some point.